content/docs/installation/best-practice.md

```diff
@@ -22,6 +22,201 @@ are designed for backwards compatibility rather than for best practice or maximu
+> there will be a delay while a new Pod is started on another Node,
+> and any cert-manager resources that are created or changed during that time will not be reconciled until the new Pod starts up.
+> But the controller manager works asynchronously anyway, so any applications which depend on the cert-manager custom resources
+> will be designed to tolerate this situation.
+> That being said, the best practice is to run 2 or more replicas of each controller if the cluster has sufficient resources.
+>
+> 📖 Read [Ensure control plane stability when using webhooks](https://cloud.google.com/kubernetes-engine/docs/how-to/optimize-webhooks)
+> in the Google Kubernetes Engine (GKE) documentation,
+> for examples of how webhook disruptions might disrupt your cluster.
+
+### Topology Spread Constraints
+
+Ensure that a disruption of a node or data center does not degrade the operation of cert-manager,
+by configuring [Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/)
+for each of the components.
+For example, the following Helm chart values add topology spread constraints for all three long-running components,
+to request (but not require) Kubernetes to avoid scheduling Pods of the same Deployment to the same zone or node.
+
+```yaml
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: topology.kubernetes.io/zone
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+- maxSkew: 1
+  topologyKey: kubernetes.io/hostname
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+
+webhook:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
+
+cainjector:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: cainjector
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: cainjector
+```
+
+### PodDisruptionBudget
+
+For high availability you should also deploy a `PodDisruptionBudget` resource,
+with `minAvailable=1` *or* with `maxUnavailable=1`.
```

Comment:
I'm not sure about `maxUnavailable=1`, what happens if you only have 1 replica running? Is it OK to remove that replica in that situation?
Also, if you have a lot of replicas, `maxUnavailable > 1` might be ok too.

Reply:
`maxUnavailable: 1` requires you to have 2 running, otherwise you're stuck. Speaking from experience.

Reply:
Both settings require at least 2 replicas which is why Kyverno omit the PodDisruptionBudget if Replicas < 2

https://github.com/kyverno/kyverno/blob/53eab3d777d3dc33a1cfac73ea069c910e4b3f45/charts/kyverno/templates/background-controller/poddisruptionbudget.yaml#L1-L16

I've added a warning note about this.

I think `maxUnavailable > 1` might ok for the webhook if you have 3 replicas and you want to upgrade Nodes 2 at a time.
I'll just recommend `minAvailable=1` for the sake of simplicity.